File: Source/Extensions/Tessa.Extensions.Default.Client/Files/ClientKrPermissionsGetFileContentExtension.cs
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Cards.Extensions;
using Tessa.Extensions.Default.Shared.Workflow.KrProcess;
using Tessa.UI;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.Files
{
    public sealed class ClientKrPermissionsGetFileContentExtension :
        CardGetFileContentExtension
    {
        #region Base Overrides

        public override Task BeforeRequest(ICardGetFileContentExtensionContext context)
        {
            ICardEditorModel editor = UIContext.Current.CardEditor;
            ICardModel model;
            KrToken token;

            if (editor != null
                && (model = editor.CardModel) != null
                && (token = KrToken.TryGet(model.Card.Info)) != null
                && context.Request.VersionRowID != CardHelper.ReplacePlaceholdersVersionRowID
                && model.Card.ID == context.Request.CardID)
            {
                token.Set(context.Request.Info);
            }

            return Task.CompletedTask;
        }

        #endregion
    }
}

File: WebClient SDK/src/default/files/clientKrPermissionsGetFileContentExtension.ts
import {
  CardGetFileContentExtension,
  ICardGetFileContentExtensionContext
} from 'tessa/cards/extensions';
import { UIContext } from 'tessa/ui';
import { ICardModel } from 'tessa/ui/cards';
import { KrToken } from 'tessa/workflow';
import { CardTemplateHelper } from 'tessa/cards';

export class ClientKrPermissionsGetFileContentExtension extends CardGetFileContentExtension {
  public beforeRequest(context: ICardGetFileContentExtensionContext) {
    const editor = UIContext.current.cardEditor;
    let model: ICardModel;
    let token: KrToken;

    if (
      editor &&
      (model = editor.cardModel!) &&
      (token = KrToken.tryGet(model.card.info)!) &&
      context.request.versionRowId !== CardTemplateHelper.ReplacePlaceholdersVersionRowID &&
      model.card.id === context.request.cardId
    ) {
      token.setInfo(context.request.info);
    }
  }
}
